blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -141,18 +148,55 @@ export class WorkspaceSearch {
     const inputWrapper = document.createElement('div');
     Blockly.utils.dom.addClass(inputWrapper, 'ws-search-input');
     this.textInput_ = this.createTextInput_();
-    inputWrapper.append(this.textInput_);
+    if (this.textInput_) {
+      this.textInput_.addEventListener('keydown',
+          evt => this.onKeyDown_(/** @type {KeyboardEvent} */ evt));
+      this.textInput_.addEventListener('input', () => this.onInput_());
+      this.textInput_.addEventListener('click',
+          () => this.searchAndHighlight(this.preserveSelected));
+      inputWrapper.append(this.textInput_);
+    }
     searchContent.append(inputWrapper);
 
-    const actionDiv = this.createActionsDiv_();
-    searchContent.append(actionDiv);
-    searchContainer.append(this.createCloseBtn_());
+    this.actionDiv_ = document.createElement('div');
+    Blockly.utils.dom.addClass(this.actionDiv_, 'ws-search-actions');
+    searchContent.append(this.actionDiv_);
+
+    const nextBtn = this.createNextBtn_();
+    if (nextBtn) {
+      this.addBtnListener_(nextBtn, () => this.next());
+      this.actionDiv_.append(nextBtn);
+    }
+
+    const previousBtn = this.createPreviousBtn_();
+    if (previousBtn) {
+      this.addBtnListener_(previousBtn, () => this.previous());
+      this.actionDiv_.append(previousBtn);
+    }
+
+    const closeBtn = this.createCloseBtn_();
+    if (closeBtn) {
+      this.addBtnListener_(closeBtn, () => this.close())
+      searchContainer.append(closeBtn);
+    }
 
     this.HtmlDiv_.append(searchContainer);
 
     parentSvg.parentNode.insertBefore(this.HtmlDiv_, parentSvg);
   }
 
+  /**
+   * Add a button to the action div. This must be called after the init function
+   * has been called.
+   * @param {HTMLButtonElement} btn The button to add the event listener to.
+   * @param {!Function} onClickFn The function to call when the user clicks on 
```

Comment:
I think it might make sense to allow for this to be passed as null if someone wanted to just append something to the div.

Reply:
As discussed offline and in a comment before, not implementing for this pass.